src/Sentry/Platforms/Android/Extensions/SentryEventExtensions.cs

```diff
@@ -17,6 +17,12 @@ internal static class SentryEventExtensions
 
     public static SentryEvent ToSentryEvent(this JavaSdk.SentryEvent sentryEvent, JavaSdk.SentryOptions javaOptions)
     {
+        if (sentryEvent.Sdk != null)
+        {
+            // when we cast this serialize this over, this value must be set
+            sentryEvent.Sdk.Name ??= "Sentry";
+            sentryEvent.Sdk.Version ??= "1.0.0";
```

Comment:
```suggestion
            sentryEvent.Sdk.Name ??= Constants.SdkName;
            sentryEvent.Sdk.Version ??= SdkVersion.Instance.Version;
```

This is how it's done [in the Enricher](https://github.com/getsentry/sentry-dotnet/blob/ee62bb942b6bebc6b835130a5192aa68a31b6f54/src/Sentry/Internal/Enricher.cs#L61-L68)...

Reply:
This probably isn't the best place long term.  I could move it down to the beforesend logic, but anything that calls into this right now will fail if those values aren't set for one reason or another.